modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/facet/CompositeFacetProcessor.java

```diff
@@ -64,25 +63,26 @@ public AggregationBuilder processFacet(Facet facet) {
 	@Reference(
 		cardinality = ReferenceCardinality.MULTIPLE,
 		policy = ReferencePolicy.DYNAMIC,
-		policyOption = ReferencePolicyOption.GREEDY,
-		target = "(&(class.name=*)(!(class.name=DEFAULT)))"
+		policyOption = ReferencePolicyOption.GREEDY
 	)
 	protected void setFacetProcessor(
-		FacetProcessor<SearchRequestBuilder> facetProcessor,
-		Map<String, Object> properties) {
+		FacetProcessor<SearchRequestBuilder> facetProcessor) {
 
-		String className = MapUtil.getString(properties, "class.name");
+		String facetClassName = facetProcessor.getFacetClassName();
 
-		_facetProcessors.put(className, facetProcessor);
+		if (facetClassName != null) {
```

Comment:
@W2jx How could facetClassName be null?

Reply:
@tinatian It's actually a middle state, this check is removed because this class is inlined in a later commit. @W2jx  The commit order, or in another word, the order that you make these changes, should be rearranged to avoid such misunderstanding. Let's do the inline first.